src/views/AverageWaveforms/AverageWaveformsView.tsx

```diff
@@ -19,35 +23,103 @@ const AverageWaveformsView: FunctionComponent<Props> = ({data, width, height}) =
     const setSelectedPlotKeys = useCallback((keys: string[]) => {
         setSelectedUnitIds(keys.map(k => (Number(k))))
     }, [setSelectedUnitIds])
+
+    const [ampScaleFactor, setAmpScaleFactor] = useState<number>(1)
+    const [waveformsMode, setWaveformsMode] = useState<string>('geom')
+
     const plots = useMemo(() => (data.averageWaveforms.sort((a1, a2) => (a1.unitId - a2.unitId)).map(aw => ({
         key: `${aw.unitId}`,
         label: `Unit ${aw.unitId}`,
         labelColor: colorForUnitId(aw.unitId),
         props: {
             channelIds: aw.channelIds,
             waveform: subtractChannelMeans(aw.waveform),
+            layoutMode: waveformsMode,
             channelLocations: data.channelLocations,
             samplingFrequency: data.samplingFrequency,
             noiseLevel: data.noiseLevel,
+            ampScaleFactor,
             width: 120,
             height: 120
         }
-    }))), [data.averageWaveforms, data.channelLocations, data.samplingFrequency, data.noiseLevel])
-    const divStyle: React.CSSProperties = useMemo(() => ({
-        width: width - 20, // leave room for the scrollbar
-        height,
-        position: 'relative',
-        overflowY: 'auto'
-    }), [width, height])
+    }))), [data.averageWaveforms, data.channelLocations, data.samplingFrequency, data.noiseLevel, waveformsMode, ampScaleFactor])
+
+    const _handleWaveformToggle = useCallback(() => {
+        setWaveformsMode(m => (m === 'geom' ? 'vertical' : 'geom'))
+    }, [])
+    const _handleScaleAmplitudeUp = useCallback(() => {
+        setAmpScaleFactor(x => (x * 1.2))
+    }, [])
+    const _handleScaleAmplitudeDown = useCallback(() => {
+        setAmpScaleFactor(x => (x / 1.2))
+    }, [])
+    const _handleResetAmplitude = useCallback(() => {
+        setAmpScaleFactor(1)
+    }, [])
+    
+    const scalingActions = useMemo(() => {
+        const actions: ViewToolbarAction[] = [
+            {
+                type: 'button',
+                callback: _handleScaleAmplitudeUp,
+                title: 'Scale amplitude up [up arrow]',
+                icon: <FaArrowUp />,
+                keyCode: 38
+            },
+            {
+                type: 'button',
+                callback: _handleResetAmplitude,
+                title: 'Reset scale amplitude',
+                icon: <FaRegTimesCircle />
+            },
+            {
+                type: 'button',
+                callback: _handleScaleAmplitudeDown,
+                title: 'Scale amplitude down [down arrow]',
+                icon: <FaArrowDown />,
+                keyCode: 40
+            },
+            {
+                type: 'text',
+                title: 'Zoom level',
+                content: ampScaleFactor,
+                contentSigFigs: 2
+            },
```

Comment:
I'd suggest incorporating some of the changes I'd made to make the toolbar into a shared component--see the `14-toolbar-synthesis` branch of figurl, specifically [the AmplitudeScaleToolbarEntries](https://github.com/magland/figurl/blob/cba9c168ea061fa7b56a525229a40cdc2eda4bc1/src/plugins/sortingview/gui/extensions/common/sharedToolbarSets/AmplitudeScaleToolbarEntries.tsx) and [AverageWaveformsView that consumes it](https://github.com/magland/figurl/blob/cba9c168ea061fa7b56a525229a40cdc2eda4bc1/src/plugins/sortingview/gui/extensions/averagewaveforms/AverageWaveformsView/AverageWaveformsView.tsx#L56-L75).

This should make it easier to have a consistent interface across places we want to implement the same action (scaling in this case, but I also made one for time widget operations).